### ISSUE TITLE
Design syntax for customizing mark's interpolate, orientation, and stack

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#1882]
Previously users can always modify directly scale/axis/legend properties, but sometimes need to  use mark config to set properties of a mark.  We want to have a more discoverable place for all mark properties previously "hidden" in `config.mark.*`.

In Vega-Lite 2.0, we will support composition of single view plots to create multi-view and layered plots.  We plan to make `config` only a top-level config for theming and let users directly modify non-config properties to customize things.  Given that we will not make config a part of unit specification (of a single plot) in a composite spec, this "more discoverable place" for mark properties become even more needed.  

## New Encoding Channels

For mark properties that can vary across different marks in a unit spec such as `strokeDash`, they can become new encoding channels with proper field mapping support.    

```TS
{
   mark: 'line',
   encoding: {
       ...
      strokeDash: {field: ..., ... } or {value: ...}
   }
}
```

## Orient / Interpolate / Tension

However, some properties (`orient`, `interpolate`, and `tension`) should be consistent for all marks of the same unit plots and thus should not be an encoding channel.  

For orient / interpolate / tension, we are thinking about an interface like this: 

```TS
type Mark = MarkType | MarkDef;

interface MarkDef {
   mark: MarkType;

   orient: Orient; // one of 'horizontal' | 'veritical'

   // line and area
   interpolate: Interpolate; // Different line interpolator name
   tension: number;
}
```

so a mark can be a string like before, or can be an object.  For example, if users want to customize line's interpolate and tension

```
{
  mark: {
     type: 'line',
     interpolate: 'basis',
     tension: 1
  },
  ...
}
```
## Stack

For `stack`, we are thinking about adding it as a property of x and y encoding channels (for describing different stack offset (`"zero", "center", "normalize", null`).

For example, to make a stacked chart with center baseline (streamgraph), one can use an object form of a mark definition:

```
{
  "mark": "area",
  "encoding": {
    "x": {"timeUnit": "yearmonth", "field": "date", "type": "temporal"},
    "y": {"aggregate": "sum", "field": "count","type": "quantitative", stack: "center"},
    "color": {"field":"series", "type":"nominal"}
  }
}
```

This will be equilvalent to using stack config as in the [VL1 streamgraph example](http://vega.github.io/vega-editor/?mode=vega-lite&spec=stacked_area_stream).

related to #1842

cc: @mattwchun

**Comment by domoritz** [comment]
LGTM. I want to see all the properties that potentially go in there, though.

**Comment by kanitw** [comment]
Fixed in #1932
